nservicebus/sagas/saga-finding.md

```diff
@@ -21,7 +21,8 @@ snippet:nhibernate-saga-finder
 
 snippet:ravendb-saga-finder
 
+Starting with version 6 all handlers must return a `Task`, a completed `Task` or be marked `async`. Returning null will throw an `InvalidOperationException`.
 
-You can have as many of these classes as you want for a given saga or message type. If a saga can't be found, return null, and if the saga specifies that it is to be started for that message type, NServiceBus will know that a new saga instance is to be created. The above example uses NServiceBus extension for NHibernate that allows both framework and user code to share the same NHibernate session. Similar extension point exists for RavenDB.
+You can have as many of these classes as you want for a given saga or message type. If a saga can't be found, return null (or `Task.FromResult(null)` starting with version 6), and if the saga specifies that it is to be started for that message type, NServiceBus will know that a new saga instance is to be created. The above example uses NServiceBus extension for NHibernate that allows both framework and user code to share the same NHibernate session. Similar extension point exists for RavenDB.
```

Comment:
A few minor changes:
- As these are the NServiceBus docs, we usually don't specify NServiceBus when referring to specific versions
- Version should be capitalized.
- I tried to simplify it a bit.

> If a saga can't be found return `null`. In Version 6 and above, if the method is not marked `async`, return `Task.FromResult(null)`.